src/model.rs

```diff
@@ -422,7 +430,18 @@ impl Model<ProblemCreated> {
 
 impl Model<Solved> {
     fn _set_best_sol(&mut self) {
-        if self.scip.get_status() == Status::Optimal {
+        // a best solution can exist even if the status is not optimal, for example if the 
+        // time limit is reached. In this case, the best solution is the best solution explored,
+        // assuming any viable solutions had been explored.
+
+        // to check if any viable solutions had been explored and to fetch the best solution,
+        // the problem needs to be in a stage where it has variables and constraints.
+        if !self.scip.has_vars_and_constraints() {
```

Comment:
Is this needed? It should be prevented from russcip's type state. The solve method can only be called from `Model<ProblemCreated>` state.

Reply:
I'm not as familiar with SCIP stages and I couldn't find documentation on their relationship with russcip states, so preferred to err on the safe sided. 

I gather from the comment that the calls to `get_n_sols` and `get_best_sol` should be safe in Model<ProblemCreated>, will revert that check.

Reply:
reverted